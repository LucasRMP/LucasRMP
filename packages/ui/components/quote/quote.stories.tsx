import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'

import { Quote } from './quote'

export default {
  title: 'ui/Quote',
  component: Quote,
  argTypes: {
    children: {
      type: { name: 'string', required: true },
      description: 'The text to display inside the button',
      defaultValue: 'This is a quote from someone very wise',
    },
  },
} as ComponentMeta<typeof Quote>

const Template: ComponentStory<typeof Quote> = args => <Quote {...args} />

export const Primary = Template.bind({})
