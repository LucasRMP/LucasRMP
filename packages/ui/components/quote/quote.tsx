import React from 'react'

export const Quote: React.FC<React.PropsWithChildren> = ({ children }) => (
  <blockquote className="text-xl text-slate-300 border-l-4 border-primary px-3 py-1">
    <p>{children}</p>
  </blockquote>
)
