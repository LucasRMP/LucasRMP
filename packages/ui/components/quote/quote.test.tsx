import React from 'react'
import { render, screen } from '@testing-library/react'

import { Quote } from './quote'

describe('ui/Quote', () => {
  it('should render a blockquote element with the children passed in', () => {
    const { container } = render(
      <Quote>
        <span data-testid="children"></span>
      </Quote>,
    )

    expect(screen.getByTestId('children')).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
