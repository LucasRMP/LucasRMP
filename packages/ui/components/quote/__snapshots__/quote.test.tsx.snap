// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/Quote should render a blockquote element with the children passed in 1`] = `
<div>
  <blockquote
    class="text-xl text-slate-300 border-l-4 border-primary px-3 py-1"
  >
    <p>
      <span
        data-testid="children"
      />
    </p>
  </blockquote>
</div>
`;
